Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -38,6 +38,11 @@
 
 namespace IronPython.Runtime.Operations {
 
+    public class ExceptionMetadata {
```

Comment:
Maybe make this internal? Does this need to be mutable (i.e. can we get rid of the setters and use constructors instead)?

Reply:
Does not need to mutable. In order to make it internal `SaveCurrentException` and `RestoreCurrentException` will need to be internal as well. That's probably what they should be anyways.

Reply:
Ah right. I think those need to be public to work with compiled code.